src/app/clusters/time-format-localization-server/time-format-localization-server.cpp

```diff
@@ -166,38 +166,40 @@ static Protocols::InteractionModel::Status emberAfPluginTimeFormatLocalizationOn
     return res;
 }
 
-static Protocols::InteractionModel::Status
-emberAfPluginTimeFormatLocalizationOnUnhandledAttributeChange(EndpointId EndpointId, EmberAfAttributeType attrType,
-                                                              uint16_t attrSize, uint8_t * attrValue)
-{
-    return Protocols::InteractionModel::Status::Success;
-}
-
 Protocols::InteractionModel::Status MatterTimeFormatLocalizationClusterServerPreAttributeChangedCallback(
     const ConcreteAttributePath & attributePath, EmberAfAttributeType attributeType, uint16_t size, uint8_t * value)
 {
-    Protocols::InteractionModel::Status res;
-
     switch (attributePath.mAttributeId)
     {
-    case ActiveCalendarType::Id:
-        if (sizeof(uint8_t) == size)
-        {
-            res = emberAfPluginTimeFormatLocalizationOnCalendarTypeChange(attributePath.mEndpointId,
-                                                                          static_cast<CalendarTypeEnum>(*value));
-        }
-        else
-        {
-            res = Protocols::InteractionModel::Status::InvalidValue;
-        }
-        break;
+    case ActiveCalendarType::Id: {
+        VerifyOrReturnValue(sizeof(uint8_t) == size, Protocols::InteractionModel::Status::InvalidValue);
 
-    default:
-        res = emberAfPluginTimeFormatLocalizationOnUnhandledAttributeChange(attributePath.mEndpointId, attributeType, size, value);
-        break;
+        CalendarTypeEnum calendarType = static_cast<CalendarTypeEnum>(*value);
+
+        // Valid range is from 0 to unknown. This relies that unknown value is
+        // the first unused value and values increase over time
```

Comment:
We shouldn't rely on that.  That's subject to change (and people have been wanting to change it).

This check should be using only actual spec-defined values.

Reply:
Updated - it seems IM validates valid enum rances, so a check for kUnknown makes return values correct.

I am surprised we did not return errors immediately for any kUnknown (since we convert values anyway).

Reply:
Hard to re-work this - kUnknown was actually not working and only calendar types had extra checks. Hard-coded available enum values.